src/libs/actions/BankAccounts.js

```diff
@@ -314,115 +315,131 @@ function goToWithdrawalAccountSetupStep(stepID, achData) {
-                        // Expensify card automatically. However, we will still handle that step and show the Validate
-                        // view.
-                        currentStep = CONST.BANK_ACCOUNT.STEP.ENABLE;
+                    // Ensure we route the user to the correct step based on the status of their bank account
+                    if (bankAccount && !currentStep) {
+                        currentStep = bankAccount.isPending() || bankAccount.isVerifying()
+                            ? CONST.BANK_ACCOUNT.STEP.VALIDATION
+                            : CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT;
+
+                        // @TODO Again, not sure how much of this logic is needed right now as we shouldn't be handling any
+                        // open accounts in E.cash yet that need to pass any more checks or can be upgraded, but leaving in for
+                        // possible future compatibility.
+                        if (bankAccount.isOpen()) {
+                            if (bankAccount.needsToPassLatestChecks()) {
+                                const hasTriedToUpgrade = bankAccount.getDateSigned()
+                                    > (kycVerificationsMigration || '2020-01-13');
+                                currentStep = hasTriedToUpgrade
+                                    ? CONST.BANK_ACCOUNT.STEP.VALIDATION : CONST.BANK_ACCOUNT.STEP.COMPANY;
+                                achData.bankAccountInReview = hasTriedToUpgrade;
+                            } else {
+                                // In Expensify.cash we do not show a specific view for the EnableStep since we will enable the
+                                // Expensify card automatically. However, we will still handle that step and show the Validate
+                                // view.
+                                currentStep = CONST.BANK_ACCOUNT.STEP.ENABLE;
+                            }
+                        }
                     }
-                }
-            }
 
-            // If at this point we still don't have a current step, default to the BankAccountStep
-            if (!currentStep) {
-                currentStep = CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT;
-            }
+                    // If at this point we still don't have a current step, default to the BankAccountStep
+                    if (!currentStep) {
+                        currentStep = CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT;
+                    }
 
-            Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {throttledDate});
-            goToWithdrawalAccountSetupStep(currentStep, achData);
-        })
-        .finally(() => {
-            Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {loading: false});
+                    // Reset Error key
+                    const error = '';
+                    Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {throttledDate, failedValidationAttempts, error});
```

Comment:
NAB, can inline this instead of `const error`

```js
Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {throttledDate, failedValidationAttempts, error: ''});
```